### PR TITLE
[DCJ-400] Minor and patch dependency updates, inc. Spring Boot 3.2.4 -> 3.3.2

### PR DESCRIPTION
**Pull request by okotsopoulos**
__Jira ticket__: https://broadworkbench.atlassian.net/browse/DCJ-400

## Addresses

Minor and patch dependency updates, which should take care of most of the updates proposed in https://github.com/DataBiosphere/jade-data-repo/pull/1763 (TDR's first grouped Dependabot PR) and triangulate on remaining troublesome ones.

I broke up my changes into logical commits for easier review.

## Summary of changes

| Package | From | To |
| --- | --- | --- |
| [com.google.cloud:libraries-bom](https://github.com/googleapis/java-cloud-bom) | `26.30.0` | `26.43.0` |
| [org.liquibase:liquibase-core](https://github.com/liquibase/liquibase) | `4.26.0` | `4.29.1` |
| [org.antlr:ST4](https://github.com/antlr/stringtemplate4) | `4.3` | `4.3.4` |
| net.javacrumbs.shedlock:shedlock-provider-jdbc-template | `5.2.0` | `5.14.0` |
| [net.javacrumbs.shedlock:shedlock-spring](https://github.com/lukas-krecan/ShedLock) | `5.2.0` | `5.14.0` |
| bio.terra:terra-policy-client | `1.0.11-SNAPSHOT` | `1.0.15-SNAPSHOT` |
| bio.terra:externalcreds-client-resttemplate | `1.3.0-SNAPSHOT` | `1.45.0-SNAPSHOT` |
| [com.fasterxml.jackson.core:jackson-core](https://github.com/FasterXML/jackson-core) | `2.15.3` | [2.17.2](https://github.com/spring-projects/spring-boot/blob/a4c428a328d969fb963074f25981f5fc30b34bfd/gradle.properties#L12) (now versioned by Spring dependency manager) |
| [com.fasterxml.jackson.core:jackson-annotations](https://github.com/FasterXML/jackson) | `2.15.3` | [2.17.2](https://github.com/spring-projects/spring-boot/blob/a4c428a328d969fb963074f25981f5fc30b34bfd/gradle.properties#L12) (now versioned by Spring dependency manager) |
| [com.fasterxml.jackson.core:jackson-databind](https://github.com/FasterXML/jackson) | `2.15.3` | [2.17.2](https://github.com/spring-projects/spring-boot/blob/a4c428a328d969fb963074f25981f5fc30b34bfd/gradle.properties#L12) (now versioned by Spring dependency manager) |
| [com.azure:azure-identity](https://github.com/Azure/azure-sdk-for-java) | `1.11.1` | `1.13.2` |
| [com.azure.resourcemanager:azure-resourcemanager](https://github.com/Azure/azure-sdk-for-java) | `2.34.0` | `2.41.0` |
| [com.azure:azure-storage-common](https://github.com/Azure/azure-sdk-for-java) | `12.24.1` | `12.26.0` |
| [com.azure:azure-storage-file-datalake](https://github.com/Azure/azure-sdk-for-java) | `12.18.1` | `12.20.0` |
| [com.azure:azure-data-tables](https://github.com/Azure/azure-sdk-for-java) | `12.3.18` | `12.4.3` |
| [io.opentelemetry.instrumentation:opentelemetry-instrumentation-annotations](https://github.com/open-telemetry/opentelemetry-java-instrumentation) | `2.2.0` | `2.6.0` |
| [org.apache.parquet:parquet-common](https://github.com/apache/parquet-mr) | `1.12.0` | `1.14.1` |
| [org.apache.parquet:parquet-hadoop](https://github.com/apache/parquet-mr) | `1.12.0` | `1.14.1` |
| [org.apache.parquet:parquet-hadoop-bundle](https://github.com/apache/parquet-mr) | `1.12.0` | `1.14.1` |
| [org.apache.parquet:parquet-encoding](https://github.com/apache/parquet-mr) | `1.12.0` | `1.14.1` |
| [org.apache.parquet:parquet-column](https://github.com/apache/parquet-mr) | `1.12.0` | `1.14.1` |
| org.apache.hadoop:hadoop-common | `3.3.1` | `3.4.0` |
| org.apache.hadoop:hadoop-azure | `3.3.1` | Removed (unused) |
| org.apache.hadoop:hadoop-mapreduce-client-core | `3.3.1` | `3.4.0` |
| [au.com.dius.pact.provider:junit5](https://github.com/pact-foundation/pact-jvm) | `4.3.19` | `4.6.12` |
| [au.com.dius.pact.provider:junit5spring](https://github.com/pact-foundation/pact-jvm) | `4.3.19` | `4.6.12` |
| [org.antlr:antlr4](https://github.com/antlr/antlr4) | `4.8` | `4.13.2` |
| [org.codehaus.groovy:groovy](https://github.com/apache/groovy) | `3.0.7` | Removed (unused) |
| [info.picocli:picocli](https://github.com/remkop/picocli) | `4.7.5` | `4.7.6` |
| [io.zonky.test:embedded-database-spring-test](https://github.com/zonkyio/embedded-database-spring-test) | `2.5.0` | `2.5.1` |
| [io.zonky.test:embedded-postgres](https://github.com/zonkyio/embedded-postgres) | `2.0.6` | `2.0.7` |
| com.google.cloud.tools.jib | `3.2.0` | `3.4.3` |
| org.gradle.test-retry | `1.5.8` | `1.5.10` |
| [org.springframework.boot](https://github.com/spring-projects/spring-boot) | `3.2.4` | `3.3.2` |
| [io.spring.dependency-management](https://github.com/spring-gradle-plugins/dependency-management-plugin) | `1.1.4` | `1.1.6` |
| com.gorylenko.gradle-git-properties | `2.4.1` | `2.4.2` |

## Testing Strategy

- Expanded unit tests to cover modified code.
- Ran a Static Code Analysis off of this feature branch to confirm that no vulnerabilities were introduced or reintroduced: https://github.com/broadinstitute/dsp-appsec-sourceclear-github-actions/actions/runs/10292414130/job/28486770708

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -162,19 +162,13 @@ configurations {
     runtimeClasspath
 }
 
-// Spring Boot 3.2.4 pulls in opentelemetry-bom 1.31.0.
-// It must have version >= 1.34.1 for compatibility with terra-common-lib 1.1.10:
-ext['opentelemetry.version'] = '1.36.0'
```
Comment (okotsopoulos) [inline build.gradle:167]: Now [1.37.0](https://github.com/spring-projects/spring-boot/blob/v3.3.2/spring-boot-project/spring-boot-dependencies/build.gradle#L1396).

File: build.gradle [hunk 2]
```diff
@@ -162,19 +162,13 @@ configurations {
...
-// Spring Boot 3.2.4 pulls in io.netty:netty-bom 4.1.107.Final which is impacted by CVE-2024-29025.
-ext['netty.version'] = '4.1.108.Final'
```
Comment (okotsopoulos) [inline build.gradle:169]: Now [4.1.111.Final](https://github.com/spring-projects/spring-boot/blob/v3.3.2/spring-boot-project/spring-boot-dependencies/build.gradle#L1379).

File: build.gradle [hunk 3]
```diff
@@ -201,90 +197,75 @@ dependencies {
...
-    }
-
-    testImplementation 'au.com.dius.pact.provider:junit5:4.3.19'
-    testImplementation 'au.com.dius.pact.provider:junit5spring:4.3.19'
 
-    antlr 'org.antlr:antlr4:4.8'
+    testImplementation 'au.com.dius.pact.provider:junit5:4.6.12'
+    testImplementation 'au.com.dius.pact.provider:junit5spring:4.6.12'
 
-    // Need groovy on the class path for the logback config. Could use XML and skip this dependency,
```
Comment (okotsopoulos) [inline build.gradle:276]: We use XML now, and logback hasn't supported groovy since 2021: https://logback.qos.ch/news.html#1.2.9

**Review by snf2ye — APPROVED** [review]

**Review by rushtong — APPROVED** [review]
👍🏽

File: build.gradle [hunk 4]
```diff
@@ -201,90 +197,82 @@ dependencies {
...
-
-    testImplementation 'org.apache.parquet:parquet-common:1.12.0'
-    testImplementation 'org.apache.parquet:parquet-hadoop:1.12.0'
-    testImplementation 'org.apache.parquet:parquet-hadoop-bundle:1.12.0'
-    testImplementation 'org.apache.parquet:parquet-encoding:1.12.0'
-    testImplementation 'org.apache.parquet:parquet-column:1.12.0'
-    testImplementation ('org.apache.hadoop:hadoop-common:3.3.1')  {
+    implementation 'io.opentelemetry.instrumentation:opentelemetry-instrumentation-annotations:2.6.0'
+
+    testImplementation 'org.apache.parquet:parquet-common:1.14.1'
```
Comment (pshapiro4broad) [inline build.gradle:243]: For repeated values like this we could use a variable
```
dependencies {
  ext {
    parquet = '1.14.1'
  }
  testImplementation "org.apache.parquet:parquet-common:${parquet}"
}

**Review by pshapiro4broad — APPROVED** [review]

**Pull request merged**